electrum/gui/qt/main_window.py

```diff
@@ -1709,11 +1709,12 @@ def pay_onchain_dialog(
             fee=fee_est,
             is_sweep=is_sweep)
         output_values = [x.value for x in outputs]
-        if output_values.count('!') > 1:
-            self.show_error(_("More than one output set to spend max"))
-            return
-
-        output_value = '!' if '!' in output_values else sum(output_values)
+        for value in output_values:
+            if value[-1] == '!':
+                output_value = '!'
+                break
+            else:
+                output_value += value
```

Comment:
bug1: This last line `+=` does not work as `output_value` is not yet defined.
bug2: the `value[-1] == '!'` is going to raise if value is an int, which is actually the most common case

-----

Some examples:
```
tb1qk3g0t9pw5wctkzz7gh6k3ljfuukn729s67y54e, 0.002
````
^ is broken.

```
tb1qr7evucrllljtryam6y2k3ntmlptq208pghql2h, !
tb1qar4ye43tdfj6y5n3yndp9adhs2wuz2v0wgqn5l, 3!
````
^ works.

As discussed, we would like even mixed cases such as below to work:
```
tb1qk3g0t9pw5wctkzz7gh6k3ljfuukn729s67y54e, 0.002
tb1qr7evucrllljtryam6y2k3ntmlptq208pghql2h, !
tb1qs3msqp0n0qade2haanjw2dkaa5lm77vwvce00h, 0.003
tb1qar4ye43tdfj6y5n3yndp9adhs2wuz2v0wgqn5l, 3!
```
^ also broken atm.

-----

You could just do (with the modified `parse_max_spend` I describe in another comment):

```suggestion
        if any(parse_max_spend(outval) for outval in output_values):
            output_value = '!'
        else:
            output_value = sum(output_values)
```